src/Tools.pm

```diff
@@ -530,22 +554,24 @@ otherwise 0.
 
 =cut
 
-sub IsDMDevice {
-    my $dev = shift;
+sub IsDMDevice
+{
+  my $self = $lib_ref;
+  my $dev = shift;
+
+  my $r = $self->{cache}{IsDMDevice}{$dev};
```

Comment:
please use more descriptive value. I have trouble to recognize, that it is actually return value.